### PR TITLE
#160619267 Build REST API end-point to signup a user 

### PR DESCRIPTION
**Pull request by Daymorelah**
#### What does this PR do?
Create REST API end-point that registers a user on the app
#### Description of Task to be completed?
* Ensure recommended style guide is configured correctly.
* Create a route `/auth/signup` that handles registering a user on the app.
* Use postman to test this endpoint and ensure its working as expected.
* Write tests for this endpoint.
#### How should this be manually tested?
* Fork the repo and `cd` into the project's directory
* Run `npm test` to test this feature
* Run `npm run test-coverage` to get the test coverage.
#### What are the relevant pivotal tracker stories?
#160619267, #160627993

### CODE REVIEW

File: server/Controller/userController.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    const { username, password, email } = req.body;
+    let encryptedPassword;
+    CryptData.encryptData(password).then((hash) => {
+      encryptedPassword = hash;
+      pool.query(queries.signup, [username, encryptedPassword, email],
+        (error, response) => {
+          if (error) {
+            res.status(409).jsend.fail({
+              code: 409,
+              message: 'User details allready exist. Signup was not successful',
```
Comment (ChidinmaOrajiaku) [inline server/Controller/userController.js]: `allready` should be `already`

**Review by ChidinmaOrajiaku — CHANGES_REQUESTED** [review]

File: server/Controller/userController.js (hunk 1 above)
Comment (Daymorelah) [reply server/Controller/userController.js]: The typo-error has been corrected and other response messages have been checked. A PR to implement this feedback is via his link: https://github.com/Daymorelah/orderMeal/pull/27

File: server/Utilities/tokenAuth.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+   * @param {object} req - Request object 
+   * @param {object} res - Response object
+   * @param {callback} next - The callback that passes the request to the next handler
+   */
+  static checkToken(req, res, next) {
+    const token = req.body.token || req.query.token || req.headers['x-access-token'];
+    if (!token) {
+      res.status(401).jsend.fail({
+        code: 401,
+        message: 'User not auhorized',
```
Comment (ChidinmaOrajiaku) [inline server/Utilities/tokenAuth.js]: `User not authorised`

**Comment by ChidinmaOrajiaku** [comment]
Just fix the changes I requested and merge

**Pull request merged**